### PR TITLE
Convert Docker to use new SDK Resource Manager

### PR DESCRIPTION
**Pull request by mitchellh**
This converts the Docker plugin to use the new [resource manager library](https://github.com/hashicorp/waypoint-plugin-sdk/pull/29) in the SDK. This will improve error handling and makes the code a bit more structured. There is no other functionality changes in this PR.

### CODE REVIEW

**Review by krantzinator — APPROVED** [review]

**Review by izaaklauer — APPROVED** [review]

File: builtin/docker/platform.go [hunk 1]
```diff
@@ -349,12 +452,12 @@ func (p *Platform) pullImage(cli *client.Client, log hclog.Logger, ui terminal.U
 
 		// if we have images do not pull
 		if len(sum) > 0 {
+			s.Done()
```
Comment (briancain) [inline builtin/docker/platform.go:455]: I _think_ Evans recent work with stepgroups is going to fix this, but I noticed  this line never resolves as Done:

Screenshot from 2021-05-18 07-50-02

Comment (briancain) [reply builtin/docker/platform.go:455]: We could add an update here that's like:

```suggestion
			s.Update("Docker image %q up to date!", in)
			s.Done()
```

**Review by briancain — CHANGES_REQUESTED** [review]
NIce! Looks good, I just had a request to update some of the output 😄 But otherwise works well for me 🎉 🐳

**Review by briancain — APPROVED** [review]
🐳 🎉

**Pull request merged**